### PR TITLE
Stripe Coupon and Offer.is_promotional

### PR DESCRIPTION
**Pull request by rhimmelbauer**
Notes:

- Added coupon on the build_subscription in the stripe_processor.
- Invoice.upate_totals take into consideration promotional offer and their discounts.

### CODE REVIEW

File: develop/core/tests/test_invoice.py [hunk 1]
```diff
@@ -239,6 +238,27 @@ def test_order_item_no_trial_amount(self):
         self.assertEqual(self.new_invoice.get_discounts(), 0)
```
Comment (devo-wm) [inline develop/core/tests/test_invoice.py:238]: i love your tests. they tell us exactly what's changed.

File: develop/core/tests/test_invoice.py [hunk 2]
```diff
@@ -239,6 +238,27 @@ def test_order_item_no_trial_amount(self):
...
+        month_offer = Offer.objects.get(pk=6)
+        self.new_invoice.global_discount = 10
+        self.new_invoice.save()
+        self.new_invoice.add_offer(month_offer)
+        self.assertEqual(self.new_invoice.total, month_offer.current_price() - self.new_invoice.global_discount)
+
+    def test_invoice_negative_global_discount(self):
+        month_offer = Offer.objects.get(pk=6)
+        self.new_invoice.global_discount = -10
+        self.new_invoice.save()
```
Comment (devo-wm) [inline develop/core/tests/test_invoice.py:251]: cool, but fmi - what's the use case for a **negative** discount?

File: develop/core/tests/test_invoice.py [hunk 3]
```diff
@@ -239,6 +238,27 @@ def test_order_item_no_trial_amount(self):
...
+        self.new_invoice.add_offer(month_offer)
+        self.assertEqual(self.new_invoice.total, month_offer.current_price() + self.new_invoice.global_discount)
+    
+    def test_invoice_global_discount_no_less_than_zero(self):
+        month_offer = Offer.objects.get(pk=6)
+        self.new_invoice.add_offer(month_offer)
+        self.new_invoice.global_discount = 100
```
Comment (devo-wm) [inline develop/core/tests/test_invoice.py:258]: would it be better to test with 
`self.new_invoice.global_discount = 101`
?

**Comment by rhimmelbauer** [comment]
> cool, but fmi - what's the use case for a **negative** discount? 

To check that math.fabs() is doing it's job. Don't want to get caught in a scenario where you might endup adding or substracting more depending on what the user inputed. It should work in either positive or negative values

File: develop/core/tests/test_invoice.py (hunk 2 above)
Comment (rhimmelbauer) [reply develop/core/tests/test_invoice.py:251]: To check that math.fabs() is doing it's job. Don't want to get caught in a scenario where you might endup adding or substracting more depending on what the user inputed. It should work in either positive or negative values

**Review by devo-wm — APPROVED** [review]

**Pull request merged**